### PR TITLE
[OTX] Multi-GPU: run a thread to check parent process alive when child process starts

### PR DESCRIPTION
**Pull request by eunwoosh**
### Summary

- run a thread to check a parent process alive when child process starts

### CODE REVIEW

**Comment by eunwoosh** [comment]
I checked that all multi GPU tests are passed.
multi GPU test result

File: requirements/base.txt [hunk 1]
```diff
@@ -19,3 +19,4 @@ scikit-learn==0.24.*
 Shapely>=1.7.1,<=1.8.0
 hpopt==0.3.0
 imagesize==1.4.1
+psutil
```
Comment (goodsong81) [inline requirements/base.txt:23]: BSD 3-Clause "New" or "Revised" License.
https://github.com/giampaolo/psutil/blob/master/LICENSE

Looks OK but @yunchu could you check this?

File: otx/cli/utils/multi_gpu.py [hunk 2]
```diff
@@ -237,8 +238,23 @@ def run_child_process(
...
+        threading.Thread(target=MultiGPUManager.check_parent_processes_alive, daemon=True).start()
+
         train_func()
 
+    @staticmethod
+    def check_parent_processes_alive():
+        """Check parent process is alive and if not, exit by itself."""
+        cur_process = psutil.Process()
+        parent = cur_process.parent()
+        while True:
```
Comment (jaegukhyun) [inline otx/cli/utils/multi_gpu.py:250]: I have a little concern about this. Is there no need to add a timeout?

**Review by jaegukhyun — COMMENTED** [review]
I have a comment about "while True:"

File: otx/cli/utils/multi_gpu.py (hunk 2 above)
Comment (eunwoosh) [reply otx/cli/utils/multi_gpu.py:250]: Yes. this function is always started as daemon. So, when parent process exits, then this process also exits automatically.

File: otx/cli/utils/multi_gpu.py [hunk 3]
```diff
@@ -237,8 +238,23 @@ def run_child_process(
...
+            time.sleep(1)
+            if not parent.is_running():
+                break
+
+        logger.warning("Parent processe is terminated abnormally. process exits.")
```
Comment (JihwanEom) [inline otx/cli/utils/multi_gpu.py]: Simple typo.. `processe` -> `process`, and `process exits.` -> `Process exits.` (Capital)

Comment (eunwoosh) [reply otx/cli/utils/multi_gpu.py]: Thanks for pointing. I will update it.

**Review by JihwanEom — APPROVED** [review]

File: requirements/base.txt (hunk 1 above)
Comment (goodsong81) [reply requirements/base.txt:23]: @yunchu It's a reminder. Could you please confirm when you are available?

**Review by goodsong81 — APPROVED** [review]
LGTM. I would wait for @yunchu's comment.

File: requirements/base.txt (hunk 1 above)
Comment (yunchu) [reply requirements/base.txt:23]: @goodsong81 @eunwoosh 
There is no issue to use BSD 3-Clause "New" or "Revised" License. 
The only thing required is that adding this dependency to the BOM to update OSPDT for the next release.

**Pull request merged**